### ISSUE TITLE
Handle challenge deletion and/or name changes

### ISSUE THREAD

**Issue by pl4nty** [CTFd/ctfcli#115]
I currently use ctfcli like an infrastructure-as-code provider in CI/CD pipelines. However, the following resource changes (challenges/containers/services) must be reconciled manually:
* Name changes, as it's the sync anchor
* Resource deleted from repo

Other providers like [Azure](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/deployment-modes) offer a "complete" mode, where resources that aren't defined in the repo are deleted. This would support the change types listed above.

Not sure whether this aligns to ctfcli's design though - maybe a Terraform provider would be a better fit, since it's common for CTF infrastructure anyway.

**Comment by pl4nty** [comment]
Looks like there's a Terraform version here, not sure how stable it is though: https://github.com/PsypherPunk/terraform-provider-ctfd

**Comment by reteps** [comment]
Would be fixed by #159

**Comment by pl4nty** [comment]
in my case, my CI is stateless, so #159 wouldn't work for me

**Comment by reteps** [comment]
I don't think you could reconcile renames with a stateless system unless you diff commit history (which in itself, is state).

**Comment by reteps** [comment]
I am sortof suggesting a new paradigm:

- You use your CI pipeline to commit the .ctf/config file to git
- The CI is no longer stateless

**Comment by reteps** [comment]
Would you be able to provide more details about your usecase?

**Comment by pl4nty** [comment]
I suggested delete/recreate because I don't run ctfcli during competitions, to avoid any unexpected changes. but I understand that this approach might not align to ctfcli's design

I [generate `.ctf/config` at runtime](https://github.com/pl4nty/auto-CTFd/blob/69313ee2b57462497aac7b562f7ad2189abc2dcc/.github/workflows/challenges.yml#L28-L47) to avoid storing credentials in git, and allow challenge creation/testing without ctfcli
